generate_response.py

```diff
@@ -879,6 +909,17 @@ def description_2_code_multi_rounds(prompt, user_input, original_prompt, model,
                 continue
 
             ## 3rd round: generate final code: generate 2nd-round code with chat history (Q&A)
+            if model == "AgentCoder":
+                # # We can only send one prompt to AgentCoder for now. Adding multiple roles requires major code changes in the original AgentCoder repo
+                # new_prompt = "Original Question: " + original_prompt + " First Response: " + response + " Feedback: " + answer + " " + PROMPT_2ND_ROUND
+                # messages[-1]["prompt"] = new_prompt
+                # msgs_i = messages.copy()
+
+                # directly send third round request to GPT
+                messages.clear()
+                messages.append({"role": "user","content": full_prompt})
```

Comment:
switch to flatening this to a single string

Reply:
done